lib/jdomodels/src/main/java/org/sagebionetworks/repo/model/dbo/dao/NodeDAOImpl.java

```diff
@@ -893,41 +946,31 @@ public EntityHeader getEntityHeader(String nodeId, Long versionNumber) throws Da
 		if (md5 == null) {
 			throw new IllegalArgumentException("md5 cannot be null.");
 		}
-
 		MapSqlParameterSource paramMap = new MapSqlParameterSource();
 		paramMap.addValue(COL_FILES_CONTENT_MD5, md5);
-		List<Map<String, Object>> rowList = namedParameterJdbcTemplate.queryForList(SELECT_NODE_VERSION_BY_FILE_MD5, paramMap);
+		List<EntityHeader> rowList = namedParameterJdbcTemplate.query(SELECT_NODE_VERSION_BY_FILE_MD5, paramMap, new RowMapper<EntityHeader>() {
+
+			@Override
+			public EntityHeader mapRow(ResultSet rs, int rowNum)
+					throws SQLException {
+				EntityHeader header = new EntityHeader();
+				Long entityId = rs.getLong(COL_NODE_ID);
+				header.setId(KeyFactory.keyToString(entityId));
+				EntityType type = EntityType.valueOf((String) rs.getString(COL_NODE_TYPE));
+				header.setType(EntityTypeUtils.getEntityTypeClassName(type));
+				header.setName(rs.getString(COL_NODE_NAME));
+				header.setVersionNumber(rs.getLong(COL_REVISION_NUMBER));
+				header.setVersionLabel(rs.getString(COL_REVISION_LABEL));
+				header.setBenefactorId(rs.getLong(COL_NODE_BENEFACTOR_ID));
```

Comment:
replace version label with version number to be consistent with the earlier query, please.